src/main/java/org/example/homework4/LeapYear.java

```diff
@@ -0,0 +1,18 @@
+package org.example.homework4;
+
+public class LeapYear {
+    public static void main(String[] args) {
+        int year = 7;
+         if (leapYear(year))  {
+             System.out.println(year + " is a Leap Year");
+         } else System.out.println(year + " isn't a Leap Year or out of range [1;9999]");
+    }
+    private static boolean leapYear(int yearNumber) {
+        if (yearNumber < 1 || yearNumber > 9999) {
+            return false;
+        }
+        if (yearNumber % 4 == 0 && yearNumber % 100 != 0) {
```

Comment:
For me this logic is not obvious. And difficult to read. I suggest to combine expression to one. Like
```
        if ((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) {
            return true;
        } else return false;
```
In that case it's clearly shows what exact condition is used for considering a year as leap